web/src/peer_data.ts

```diff
@@ -262,6 +292,11 @@ export function remove_subscriber(stream_id: number, user_id: number): boolean {
     }
 
     subscribers.delete(user_id);
+    // If the sub is undefined, we'll be raising a warning in
+    // `get_loaded_subscriber_subset`, so we don't need to here.
+    if (!fetched_stream_ids.has(stream_id) && sub_store.get(stream_id)) {
+        sub_store.get(stream_id)!.subscriber_count -= 1;
+    }
```

Comment:
So there's an alternative algorithm we could take here that I think would be considerably cleaner here. Specifically, we could just recompute `subscriber_count` when we fetch full subscribers for a channel, and then just maintain `subscriber_count` from there. Benefits include:
- We don't need the conditionals here -- we just maintain `subscriber_count`.
- `get_subscriber_count` doesn't need two code paths -- it can just use `subscriber_count`, and just have a comment at the top noting it can be inaccurate if we've not fetched full subscriber data yet.
- It'd be faster to access, since we're caching the value rather than recomputing it every time.

Reply:
I don't think it would be particularly faster to access, since most of the work is iterating over users to get the bot count (which we do regardless), and getting the size of the subscriber list is constant time. But I'm happy to change it to maintain subscriber count, hoping that any code people add that changes the subscriber count makes sure to also update the count. I'll add some comments to hopefully make that clear.